components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -204,4 +230,89 @@ void DistillPageText(
   std::move(callback).Run(contents_text);
 }
 
+void DistillPageTextViaSiteScript(
+    content::RenderFrame* render_frame,
+    std::string script_content,
+    int32_t world_id,
+    base::OnceCallback<void(const std::optional<std::string>&)> callback) {
```

Comment:
nit. This may be outside the scope of this PR, but I want to leave a note that this should potentially be just `base::OnceCallback<void(std::optional<std::string>)>`. Maybe a follow up to correct this?

Reply:
The signature was copied from the pre-existing `DistillPageText` method. Since the callback value passed into DistillPageText is what we pass to DistillPageTextViaSiteScript, I kept the signature the same.

Reply:
I noticed that... that's why I was suggesting maybe a follow up, on a separate PR, but it is your choice.

Reply:
I commented on this below because this seemed very strange, but I agree since it's pre-existing there should be a follow-up issue

Reply:
I guess it isn't crazy to use const& with optional here, but it looks very weird in this context

Reply:
it is not wrong... it is just potentially obscuring a bit, and it could be causing extra copies.